File: INTRODUCCION_A_POO/EJERCICIO-9/Computadora.py
class Computadora:
    def __init__(self, procesador, ram_gb, almacenamiento_gb, tarjeta_grafica):
        self.procesador = procesador
        self.ram_gb = ram_gb
        self.almacenamiento_gb = almacenamiento_gb
        self.tarjeta_grafica = tarjeta_grafica
        self.encendida = False

    def mostrar_componentes(self):
        print("Componentes principales de la computadora:")
        print(f"Procesador: {self.procesador}")
        print(f"Memoria RAM: {self.ram_gb} GB")
        print(f"Almacenamiento: {self.almacenamiento_gb} GB")
        print(f"Tarjeta Gráfica: {self.tarjeta_grafica}")

    def estado(self):
        print("Estado de la computadora:", "Encendida" if self.encendida else "Apagada")

    def encender(self):
        if not self.encendida:
            self.encendida = True
            print("La computadora se ha encendido.")
        else:
            print("La computadora ya está encendida.")

    def apagar(self):
        if self.encendida:
            self.encendida = False
            print("La computadora se ha apagado.")
        else:
            print("La computadora ya está apagada.")

File: INTRODUCCION_A_POO/EJERCICIO-9/ComputadoraenJava/src/computadorajava/computadora.java
package computadorajava;
public class computadora {
    // Atributos principales
    private String procesador;
    private int ramGB;
    private int almacenamientoGB;
    private String tarjetaGrafica;
    private boolean encendida;

    // Constructor
    public computadora(String procesador, int ramGB, int almacenamientoGB, String tarjetaGrafica) {
        this.procesador = procesador;
        this.ramGB = ramGB;
        this.almacenamientoGB = almacenamientoGB;
        this.tarjetaGrafica = tarjetaGrafica;
        this.encendida = false;
    }

    // Método para mostrar componentes
    public void mostrarComponentes() {
        System.out.println("Componentes principales de la computadora:");
        System.out.println("Procesador: " + procesador);
        System.out.println("Memoria RAM: " + ramGB + " GB");
        System.out.println("Almacenamiento: " + almacenamientoGB + " GB");
        System.out.println("Tarjeta Gráfica: " + tarjetaGrafica);
    }

    // Método para mostrar estado
    public void mostrarEstado() {
        if (encendida) {
            System.out.println("Estado de la computadora: Encendida");
        } else {
            System.out.println("Estado de la computadora: Apagada");
        }
    }

    // Método para encender
    public void encender() {
        if (!encendida) {
            encendida = true;
            System.out.println("La computadora se ha encendido.");
        } else {
            System.out.println("La computadora ya está encendida.");
        }
    }

    // Método para apagar
    public void apagar() {
        if (encendida) {
            encendida = false;
            System.out.println("La computadora se ha apagado.");
        } else {
            System.out.println("La computadora ya está apagada.");
        }
    }
}
